test/unit/spec/narrative_core/narrativeViewersSpec.js

```diff
@@ -0,0 +1,139 @@
+        {},
+        {
+            'Viewers/a_viewer_app': {},
+            'Viewers/error_viewer_app': {
+                loading_error: 'Not a real app'
+            }
+        },
+        {},
+        {
+            'Module.Type1': {
+                view_method_ids: ['Viewers/a_viewer_app'],
+                landing_page_url_prefix: 'type1',
+                name: 'Type 1'
+            },
+            'Module.Type2': {
+                view_method_ids: ['Viewers/missing_viewer_app'],
+                landing_page_url_prefix: 'type2',
+                name: 'Type 2'
+            },
+            'Module.Type3': {
+                view_method_ids: ['Viewers/error_viewer_app'],
+                landing_page_url_prefix: 'type3',
+                name: 'Type 3'
+            }
+        }
+    ];
+    const methodData = [{
+        'Viewers/a_viewer_app': {
+            behavior: {},
+            fixed_parameters: [],
+            info: {
+                id: 'Viewers/a_viewer_app',
+                module_name: 'Viewers',
+                app_type: 'viewer',
+                input_types: ['Module.Type1'],
+                name: 'View Module Type 1',
+            },
+            parameters: [],
+            widgets: {
+                input: 'null',
+                output: 'SomeViewerWidget'
+            }
+        }
+    }];
+
+    describe('Test the NarrativeViewers module', () => {
+        beforeEach(() => {
+            jasmine.Ajax.install();
+
+            jasmine.Ajax.stubRequest(Config.url('narrative_method_store'), /list_categories/)
+                .andReturn({
+                    status: 200,
+                    statusText: 'HTTP/1 200 OK',
+                    contentType: 'application/json',
+                    responseText: JSON.stringify({
+                        version: '1.1',
+                        id: '12345',
+                        result: categoryData
+                    })
+                });
+
+            jasmine.Ajax.stubRequest(Config.url('narrative_method_store'), /get_method_spec/)
+                .andReturn({
+                    status: 200,
+                    statusText: 'HTTP/1 200 OK',
+                    contentType: 'application/json',
+                    responseText: JSON.stringify({
+                        version: '1.1',
+                        id: '12345',
+                        result: methodData
+                    })
+                });
+        });
+
+        afterEach(() => {
+            jasmine.Ajax.uninstall();
+        });
+
+        it('should load and have expected functions', () => {
+            expect(Viewers).toBeDefined();
+
+            ['getViewerInfo', 'createViewer', 'defaultViewer'].forEach(fn => {
+                expect(Viewers[fn]).toBeDefined();
+            });
+        });
+
+        it('should load viewer info as a promise', () => {
+            return Viewers.getViewerInfo().then((info) => {
+                expect(info.viewers).toBeDefined();
```

Comment:
You could put these in an array and then iterate through -- saves a line and makes it easy to amend.

Reply:
Ah, sure, I didn't see that. I didn't touch the older tests, just migrated to a different spot. I'll update.